### DISCUSSION TITLE
When app is killed from background or not opened then clicked on notification how can i redirect it on paritucal screen

### DISCUSSION THREAD

**Discussion by newbie-inReact** [invertase/react-native-firebase#5039] [Q&A]
Question
1. When the app is killed from a background or not opened, you will receive any notification & click on it -> it should redirect to that particular screen only currently it's opening the app only 

**Platform:**  Andorid & iOS

Build Info:

- "react": "16.13.1",
- "react-native": "0.63.2"

- "@react-native-firebase/firestore": "^6.4.0",
- "@react-native-firebase/messaging": "^10.3.1",
- "@react-native-firebase/perf": "^7.4.9",
- "@react-native-firebase/analytics": "^7.6.8",
- "@react-native-firebase/app": "^8.4.6",
- "@react-native-firebase/auth": "^9.3.1",
- "@react-native-firebase/crashlytics": "^8.4.11",
- "@react-native-firebase/database": "^6.7.1",
- "@react-native-firebase/firestore": "^6.4.0",
- "@react-native-firebase/messaging": "^10.3.1",
- "@react-native-firebase/perf": "^7.4.9",

```
messaging().onNotificationOpenedApp(async (remoteMessage) => {
      console.log(
          'Notification caused app to open from quit state:',
          remoteMessage,
       );
      onNotificationClick(remoteMessage.data);
      handleNotification(remoteMessage.data);
});
```

**Comment by mikehardy** [comment]
Hi there! This is not a react-native-firebase specific question I think?
In general you need some sort of state storage in your system (redux, resub, something like that)
In the click handler you store in your state storage the general idea that your user clicked on a notification, as well as anything else necessary

Now your app is booting and the UI should render etc
At some point in your UI render path you should have logic that checks state storage to see if the user has clicked on a notification, and if they have, forward

**Reply by newbie-inReact** [reply]
@mikehardy Thank you for your reply,
After a long search, I got the solution.
Firstly I've used react native splash screen so they did not provide the get notification event, Below link is the solution for that  https://github.com/invertase/react-native-firebase/issues/3469

Again Thank you for your time you can move it to closed.

**Reply by mikehardy** [reply]
please do not use react-native-splash-screen, you definitely want react-native-boot-splash, see https://github.com/crazycodeboy/react-native-splash-screen/issues/289#issuecomment-502406454
